core/templates/pages/exploration-editor-page/services/exploration-rights.service.spec.ts

```diff
@@ -325,83 +343,99 @@ describe('Exploration rights service', function() {
+        Promise.reject());
 
-      $httpBackend.expectPUT('/createhandler/status/12345')
-        .respond(500);
-      ers.publish(true).then(successHandler, failHandler);
-      $httpBackend.flush();
+      ers.publish().then(successHandler, failHandler);
+      tick();
 
       expect(clearWarningsSpy).not.toHaveBeenCalled();
       expect(successHandler).not.toHaveBeenCalled();
       expect(failHandler).toHaveBeenCalled();
-    });
+    }));
 
-  it('should save moderator change to backend', function() {
-    $httpBackend.expectPUT('/createhandler/moderatorrights/12345').respond(
-      200, sampleDataResults);
-    ers.saveModeratorChangeToBackendAsync();
-    $httpBackend.flush();
+  it('should save moderator change to backend', fakeAsync(() => {
+    spyOn(
+      explorationRightsBackendApiService,
+      'saveModeratorChangeToBackendAsyncPutData').and.returnValue(
+      Promise.resolve(serviceData));
+
+    ers.saveModeratorChangeToBackendAsync('');
+    tick();
 
     expect(clearWarningsSpy).toHaveBeenCalled();
-    expect(ers.ownerNames).toEqual(sampleDataResults.rights.owner_names);
-    expect(ers.editorNames).toEqual(sampleDataResults.rights.editor_names);
+    expect(ers.ownerNames).toEqual(
+      serviceData.rights.owner_names);
+    expect(ers.editorNames).toEqual(
+      serviceData.rights.editor_names);
     expect(ers.voiceArtistNames).toEqual(
-      sampleDataResults.rights.voice_artist_names);
-    expect(ers.viewerNames).toEqual(sampleDataResults.rights.viewer_names);
+      serviceData.rights.voice_artist_names);
+    expect(ers.viewerNames).toEqual(
+      serviceData.rights.viewer_names);
     expect(ers.isPrivate()).toEqual(true);
-    expect(ers.clonedFrom()).toEqual(sampleDataResults.rights.cloned_from);
+    expect(ers.clonedFrom()).toEqual(
+      serviceData.rights.cloned_from);
     expect(ers.isCommunityOwned()).toBe(
-      sampleDataResults.rights.community_owned);
+      serviceData.rights.community_owned);
     expect(ers.viewableIfPrivate()).toBe(
-      sampleDataResults.rights.viewable_if_private);
-  });
+      serviceData.rights.viewable_if_private);
+  }));
 
   it('should reject handler when saving moderator change to backend fails',
-    function() {
-      $httpBackend.expectPUT('/createhandler/moderatorrights/12345')
-        .respond(500);
-      ers.saveModeratorChangeToBackendAsync();
-      $httpBackend.flush();
+    fakeAsync(() => {
+      spyOn(
+        explorationRightsBackendApiService,
+        'saveModeratorChangeToBackendAsyncPutData').and.returnValue(
+        Promise.reject());
+
+      ers.saveModeratorChangeToBackendAsync('');
+      tick();
 
       expect(clearWarningsSpy).not.toHaveBeenCalled();
-      expect(ers.ownerNames).toBeUndefined();
-      expect(ers.editorNames).toBeUndefined();
-      expect(ers.voiceArtistNames).toBeUndefined();
-      expect(ers.viewerNames).toBeUndefined();
+      expect(ers.ownerNames).toBe(null);
+      expect(ers.editorNames).toBe(null);
```

Comment:
```suggestion
      expect(ers.ownerNames).toBeNull();
      expect(ers.editorNames).toBeNull();
```

please prefer to use `toBeNull` instead of `toBe(null)` here and elsewhere.

Reply:
Done

Reply:
`okay` i will use it